base/abstractarray.jl

```diff
@@ -1091,6 +1091,7 @@ zero(x::AbstractArray{T}) where {T} = fill!(similar(x), zero(T))
 # own, IndexCartesian's CartesianIndices is more complicated and requires explicit
 # inlining.
 function iterate(A::AbstractArray, state=(eachindex(A),))
+    @_propagate_inbounds_meta
     y = iterate(state...)
     y === nothing && return nothing
     A[y[1]], (state[1], tail(y)...)
```

Comment:
I thought it was `A[I]` that should be always inbounds here, always

Reply:
I thought that too, but given the kwarg `state`, I don't know if there is a chance for someone to pass one that would make `A[y[1]]` out of bounds.

Should I just mark this as `@inbounds` and remove the rest of the changes?